Brain_I2C/brain_utils.h

```diff
@@ -47,6 +60,82 @@ String make_packet(struct sensorInfo sensor) {
   return packet;
 }
 
+/*
+ * Decodes a packet sent from ground station in the following format:
+ * {<valve_ID>,<open(1) or close(0)|checksum}
+ * And calls the corresponding method in solenoids.h
+ */
+void decode_received_packet(String packet) {
```

Comment:
Instead of calling the solenoid function here, pass in a pointer to a valveInfo struct, and this function will populate that struct, and in Brain do the appropriate action

Reply:
I'm a little confused about what you mean by this. So do you mean in the loop in brain, decode what valve we are working with and pass that into this method? And what do you mean by "populate the struct"? Lastly, if Brain is doing the action, do you want this switch case statement to be in brain rather than brain_utils?